### PR TITLE
Configuration for CLI tools

### PR DESCRIPTION
**Pull request by nevir**

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import _ from 'lodash';
+
+import variantBase from './variants/_base';
+
+export const VARIANTS = [
+  'development',
+  'demo',
+  'production',
```
Comment (aowainati) [inline config/index.js:8]: I'm mildly concerned about confusion between `VARIANTS` and `NODE_ENV`. How would you distinguish the two?

File: config/variants/_base.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  // We need a monotonically increasing sequence for build identifiers. # of
+  // commits is a great proxy for that, considering that the only things that
+  // actually care about a build number are for builds against master.
+  buildNum: null,
+
+  // The endpoint to hit for API requests.
+  apiEndpoint: null,
+
+  // The S3 bucket to sync built assets to.
+  s3Bucket: null,
```
Comment (aowainati) [inline config/variants/_base.js:32]: Do these `null` keys need to be declared here? It seems like they're mainly here so they can be centrally documented, but I have a sneaking suspicion people will forget to ever update this file, and will instead just update the individual config files directly.

File: config/variants/_base.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Base configuration that all variants extend.
+ */
+export default {
+  fullCommit:  FULL_COMMIT,
+  shortCommit: SHORT_COMMIT,
+  // We need a monotonically increasing sequence for build identifiers. # of
+  // commits is a great proxy for that, considering that the only things that
+  // actually care about a build number are for builds against master.
+  buildNum: null,
```
Comment (aowainati) [inline config/variants/_base.js:26]: Do we need `buildNum`, or was that a mobile specific thing?

File: scripts/config [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+if (process.argv.includes('--all')) {
+  result = {};
+  for (const variant of config.VARIANTS) {
+    result[variant] = config.get(variant);
+  }
+} else {
+  result = config.current;
+}
+
+console.log(JSON.stringify(result, null, 2));
```
Comment (aowainati) [inline scripts/config:14]: AWWWWW SNAP ES6 SCRIPTS YESSSSS

DOWN WITH BASH

File: config/index.js (hunk 1 above)
Comment (nevir) [inline config/index.js:8]: They're the same concept.  `NODE_ENV` makes zero sense to me for a non-Node env...

File: scripts/deploy:push [hunk 5]
```diff
@@ -1,6 +1,12 @@
 #!/usr/bin/env bash
 set -e
-set -x
```
Comment (aowainati) [inline scripts/deploy:push:3]: I think deleting this from this script is the right move, but I think there are a few others (`scripts/build`) that benefit from `set -x`. Thoughts? I can't remember if you were summarily against it or not.

File: config/variants/_base.js (hunk 2 above)
Comment (nevir) [inline config/variants/_base.js:32]: Yeah, mostly for documentation - it's up to the downstream consumers to enforce their existence if they care

File: config/variants/_base.js (hunk 3 above)
Comment (nevir) [inline config/variants/_base.js:26]: We'll want it if we want client version constraints for web

File: scripts/config (hunk 4 above)
Comment (nevir) [inline scripts/config:14]: :D

File: config/index.js (hunk 1 above)
Comment (aowainati) [inline config/index.js:8]: Fair enough. We'll probably still want to make a Webpack `DefinePlugin` value for `process.env.NODE_ENV`, though, to appease weird dependencies like we did in shipper-web. I'm assuming it'll be fine to just use `variant` in that instance.

**Comment by aowainati** [comment]
image

File: scripts/deploy:push (hunk 5 above)
Comment (nevir) [inline scripts/deploy:push:3]: Ah, yeah, I'll add the rest. 

It's too spammy in the common case, especially when you're echoing things (double echo)

**Pull request merged**